aiida/common/hashing.py

```diff
@@ -338,6 +337,7 @@ def truncate_float64(x, num_bits=4):
     truncated_x = masked_int.view(np.float64)
     return truncated_x
 
+
 def truncate_array64(x, num_bits=4):
     mask = ~(2**num_bits - 1)
     int_array = np.array(x, dtype=np.float64).view(np.int64)
```

Comment:
what I forgot to ask: is there a reason to not use `numpy.ndarray.tobytes` and then hash the byte-representation? endian-ness?

Reply:
The truncation is sort of a hack to get around possibly having small changes in floating point values after storing them to the database. It's not completely safe because of course you could still have changes in the higher bits, but should remove most accidental misses. AFAICT we never really checked if this is actually needed or not.

Reply:
ah, so basically a faster `float(str(f))`